docker/CMSRucioClient/scripts/syncRolesToAttributesLocalUsers.py

```diff
@@ -44,112 +93,162 @@ def set_rse_manager(client, rse_name, site_managers, alt_rse=None):
-                    set_rse_manager(client, rse_name, site_managers, test_name)
-                    set_approvers = True
-                    break
-            if not set_approvers:
-                print("No site manager found for %s" % rse_name)
+        # group-data-manager: set of users that manage data for a physics group
+        if 'group-data-manager' in roles:
+            for tag in roles['group-data-manager']:
+                if tag.startswith('group:'):
+                    group = tag.replace("group:", '', 1)
+                    group_users[group].add(dn)
+                    dn_email_map_group[dn] = email
 
     for site, dns in local_users.items():
         try:
-            set_local_identities(client=client, site=site, dns=dns, user_map=dn_account_map, site_map=site_map)
+            # At this point site-map does not contain deleted/disabled sites
+            # (i.e it only contains site with state=ACTIVE) this is the default result from the CRIC api
+            # These sites-tags are however still present under users
+            # TODO: Should we include / exclude them?
+            if site in site_map:
+                facility = site_map[site]
+                egroup = facility_egroups[facility]
+                set_identities(client=client, account=site, egroup=egroup,
+                               target_identities=dns, dn_email_map=dn_email_map_local)
+
         except (KeyError, InvalidObject):
             print("Could not make account for %s. Perhaps the facility is not defined or the name is too long." % site)
 
+    for group, dns in group_users.items():
+        try:
+            egroup = group_egroups[group]
+            set_identities(client=client, account=group, egroup=egroup,
+                           target_identities=dns, dn_email_map=dn_email_map_group)
+
+        except (KeyError, InvalidObject):
+            print("Could not make account for %s. Perhaps the facility is not defined or the name is too long." % site)
+
+
+def set_identities(client, account, egroup, dn_email_map, target_identities=set()):
+    """
+    Creates a group account if needed
+    Updates set of assiciated identities of the group account
+    """
+
+    email = egroup + '@cern.ch'
+
+    # Check if required group account exists in Rucio
+    # If not create one
+    try:
+        print('Checking for account %s in Rucio' % account)
+        client.get_account(account)
+    except AccountNotFound:
+        print('Adding group account %s with %s' % (account, email))
+        client.add_account(account, 'GROUP', email)
+
+    # Update rucio identities to match target_identities (cric)
+    current_identities = set(identity['identity']
+                             for identity in client.list_identities(account))
+
+    add_identities = target_identities - current_identities
+    del_identities = current_identities - target_identities
+
+    for identity in add_identities:
+        print('Adding %s to %s with %s' %
```

Comment:
Since we are using python3, you should look into f-strings. It really makes things like this much more readable and less error prone.